src/Chapter3.hs

```diff
@@ -560,22 +623,31 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+
+newtype Task5Health = Task5Health Int
+newtype Armor = Armor Int
+newtype Task5Attack = Task5Attack Int
+newtype Dexterity = Dexterity Int
+newtype Strength = Strength Int
+newtype Damage = Damage Int
+newtype Task5Defense = Task5Defense Int
+
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
+    { playerHealth    :: Task5Health
+    , playerArmor     :: Armor
+    , playerAttack    :: Task5Attack
+    , playerDexterity :: Dexterity
+    , playerStrength  :: Strength
     }
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: Task5Attack -> Strength -> Damage
+calculatePlayerDamage (Task5Attack task5Attack) (Strength strength) = Damage (task5Attack + strength)
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: Armor -> Dexterity -> Task5Defense
+calculatePlayerDefense (Armor armor) (Dexterity dexterity) = Task5Defense (armor * dexterity)
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: Damage -> Task5Defense -> Task5Health -> Task5Health
+calculatePlayerHit (Damage damage) (Task5Defense defense) (Task5Health health) = Task5Health $ health + defense - damage
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype

Reply:
Thanks!
Your blog post look very informative, so I'd like to read it carefully.